stdlib/Test/src/Test.jl

```diff
@@ -263,18 +263,18 @@ struct Threw <: ExecutionResult
 end
 
 function eval_test(evaluated::Expr, quoted::Expr, source::LineNumberNode, negate::Bool=false)
-    res = true
-    i = 1
     evaled_args = evaluated.args
     quoted_args = quoted.args
     n = length(evaled_args)
     kw_suffix = ""
     if evaluated.head === :comparison
         args = evaled_args
+        res = true
+        i = 1
         while i < n
             a, op, b = args[i], args[i+1], args[i+2]
             if res
-                res = op(a, b) === true  # Keep `res` type stable
+                res = op(a, b)
```

Comment:
won't this fail on the next iteration if this isn't a bool?

Reply:
same issue below. Maybe just do an early return here if the result turns out not to be `Bool`?

Reply:
That’s the point, it now matches the behavior outside of `@test`:
```julia
julia> 1 < missing < 2
ERROR: TypeError: non-boolean (Missing) used in boolean context
```

Reply:
Returning early is wrong, we *want* an error to be thrown (`eval_test` is always run inside a `try` block).

Reply:
I could make `eval_test` return a `Threw` object in these cases instead of actually throwing (and I would do so if you think it’s better), but I don’t see the point of that. The evaluation of the test expression can throw nonetheless and I think it’s clearer to do the error handling in one place (in the expression returned by `get_test_result`).